src/api/delivery-needed/index.js

```diff
@@ -3,85 +3,105 @@ const { fields } = require("~airtable/tables/requests");
-    } catch {
-      console.error(
-        "[deliveryNeededRequestHandler] could not parse meta",
-        r.fields.meta
-      );
-    }
+  try {
+    metaJSON = JSON.parse(r.fields[meta]);
+  } catch {
+    console.error(
+      "[deliveryNeededRequestHandler] could not parse meta",
+      r.fields.meta
+    );
+  }
 
-    try {
-      const channel = metaJSON.slack_channel;
-      const timestamp = metaJSON.slack_ts;
-      slackPermalink = await slackapi.chat.getPermalink({
-        channel,
-        message_ts: timestamp
-      });
-    } catch {
-      console.error(`[deliveryNeededRequestHandler] could not fetch slack URL
+  try {
+    const channel = metaJSON.slack_channel;
+    const timestamp = metaJSON.slack_ts;
+    slackPermalink = await slackapi.chat.getPermalink({
+      channel,
+      message_ts: timestamp
+    });
+  } catch {
+    console.error(`[deliveryNeededRequestHandler] could not fetch slack URL
         for requestCode: ${r.fields[code]} channel: ${metaJSON.slack_channel} and timestamp: ${metaJSON.slack_ts}`);
-    }
+  }
 
-    return {
-      type: "Feature",
-      geometry: {
-        type: "Point",
-        coordinates: [location.lng, location.lat]
-      },
-      properties: {
-        title: r.fields[code],
-        meta: {
-          [code]: r.fields[code],
-          [crossStreetFirst]: r.fields[crossStreetFirst],
-          [crossStreetSecond]: r.fields[crossStreetSecond],
-          [neighborhoodAreaSeeMap]: r.fields[neighborhoodAreaSeeMap],
-          [firstName]: r.fields[firstName],
-          slackPermalink: slackPermalink.ok ? slackPermalink.permalink : ""
-        }
+  return {
+    type: "Feature",
+    geometry: {
+      type: "Point",
+      coordinates: [location.lng, location.lat]
+    },
+    properties: {
+      title: r.fields[code],
+      meta: {
+        [code]: r.fields[code],
+        [crossStreetFirst]: r.fields[crossStreetFirst],
+        [crossStreetSecond]: r.fields[crossStreetSecond],
+        [neighborhoodAreaSeeMap]: r.fields[neighborhoodAreaSeeMap],
+        [firstName]: r.fields[firstName],
+        [forDrivingClusters]: Boolean(r.fields[forDrivingClusters]),
+        [householdSize]: r.fields[householdSize],
+        slackPermalink: slackPermalink.ok ? slackPermalink.permalink : ""
       }
-    };
-  });
+    }
+  };
+};
+
+exports.deliveryNeededRequestHandler = async (req, res) => {
+  const [requestObj, requestErr] = await findDeliveryNeededRequests();
+
+  if (!requestObj) {
+    return res
+      .status(400)
+      .send({ message: `Error fetching requests: ${requestErr}` });
+  }
+
+  const regularRequestsPromises = requestObj
+    .filter(r => !r.fields[forDrivingClusters])
+    .map(makeFeature);
+
+  const clusterRequestsPromises = requestObj
+    .filter(r => Boolean(r.fields[forDrivingClusters]))
+    .map(makeFeature);
+
+  const regularRequests = await Promise.all(regularRequestsPromises);
+  const clusterRequests = await Promise.all(clusterRequestsPromises);
```

Comment:
fetch regular requests and cluster request separately so we can divide them up in different layers on the map.